### PR TITLE
[4.1] RavenDB-11703

### PR DESCRIPTION
**Pull request by aviv86**
http://issues.hibernatingrhinos.com/issue/RavenDB-11703

### CODE REVIEW

File: src/Raven.Server/Documents/CountersStorage.cs [hunk 1]
```diff
@@ -574,6 +577,14 @@ public string DeleteCounter(DocumentsOperationContext context, Slice key, string
 
             CreateTombstone(context, key, collection, deletedEtag, lastModifiedTicks, newEtag, newChangeVector);
 
+            context.Transaction.AddAfterCommitNotification(new DocumentChange
```
Comment (ayende) [inline src/Raven.Server/Documents/CountersStorage.cs:580]: Let's create a dedicate notification for counters.
Should include doc id, counter name and the current (overall) counter value.
Subscribing to a document should also subscribe to its counteers

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/CountersStorage.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Server/Documents/CountersStorage.cs:580]: 'Subscribing to a document should also subscribe to its counters' - how would that look from the Client API perspective? I'm not sure if we should go that way. User should either subscribe to a DocumentChange or to a CounterChange.

Comment (ppekrol) [reply src/Raven.Server/Documents/CountersStorage.cs:580]: I'm merging this. Let's handle the notification change as a separate issue.

Comment (ppekrol) [reply src/Raven.Server/Documents/CountersStorage.cs:580]: http://issues.hibernatingrhinos.com/issue/RavenDB-11727

Comment (ayende) [reply src/Raven.Server/Documents/CountersStorage.cs:580]: The idea is that when you want to hear about a document, you do it once, for all its notifications.
On the client side, you can `Subscribe(IObserver<CounterChanges>)` or not.

**Pull request merged**